lib/PersonalityProvider.jsm

```diff
@@ -38,11 +66,67 @@ this.PersonalityProvider = class PersonalityProvider {
     this.scores = scores || {};
     this.interestConfig = this.scores.interestConfig;
     this.interestVector = this.scores.interestVector;
+    this.setupSyncAttachment(RECIPE_NAME);
+    this.setupSyncAttachment(MODELS_NAME);
+  }
+
+  setupSyncAttachment(collection) {
+    RemoteSettings(collection).on("sync", async event => {
+      const {
+        data: {created, updated, deleted},
+      } = event;
+
+      // Remove every removed attachment.
+      const toRemove = deleted.concat(updated.map(u => u.old));
+      await Promise.all(toRemove.map(record => this.deleteAttachment(record)));
+
+      // Download every new/updated attachment.
+      const toDownload = created.concat(updated.map(u => u.new));
+      await Promise.all(toDownload.map(record => this.downloadAttachment(record)));
+    });
+  }
+
+  async downloadAttachment(record) {
+    const {attachment: {location, filename}} = record;
+    const headers = new Headers();
+    headers.set("Accept-Encoding", "gzip");
+    // Still not quite working. Something up with promises.
```

Comment:
Yeah, maybe `defineLazyGetter`does not work with promises... 

You may want to replace with something like:

```js
let _gBaseAttachmentURL;
async function baseAttachmentURL() {
  if (!_gBaseAttachmentURL) {
    const server = Services.prefs.getCharPref("services.settings.server");
    const serverInfo = await fetch(`${server}/`);
    const {capabilities: {attachments: {base_url}}} = serverInfo;
    _gBaseAttachmentURL = base_url;
  }
  return return _gBaseAttachmentURL;
}
```

and then: 
```js
const resp = await fetch((await baseAttachmentURL()) + location);
```

Reply:
Looks like we can still use `defineLazyGetter` async we just need to do this when we call it `const resp = await fetch((await baseAttachmentsURL) + location, {headers});`